### PR TITLE
Reset stack file to use lts-7.5 for GHC 8.0.1

### PR DESCRIPTION
**Pull request by jrbeaumont**
This change is so stack uses GHC `8.0.1` and dependencies, particularly `hint-0.6.0` as earlier versions have issues with GHC later than `7.10.3`. 

@snowleopard Please hold off on merging until I have changed the manual, README and Workcraft error messages to inform users to setup stack with the flag `--no-system-ghc`, and to point to the general install information for stack from https://docs.haskellstack.org/en/stable/install_and_upgrade/

I cannot decide, so which would you prefer:

Ask users to check their system GHC version for `>=8.0.1` and if it is, use just `stack setup` and `stack build`, otherwise use `stack setup --no-system-ghc` and `stack build --no-system-ghc`

Or

Ask all users to use `stack setup --no-system-ghc` and `stack build --no-system-ghc` regardless of their GHC version.

(With `stack build` I don't think `--no-system-ghc` changes anything, providing this flag is used for `stack setup`)

### CODE REVIEW

**Comment by snowleopard** [comment]
@jrbeaumont Let's go for `stack setup --no-system-ghc` and `stack build` unconditionally for simplicity.

**Comment by jrbeaumont** [comment]
@snowleopard: That should be all instructions updated :)

File: doc/manual.md [hunk 1]
```diff
@@ -24,48 +24,35 @@ upon the abstract, and a tool for translating concepts into *Signal
...
+Workcraft, and others. The aim of this tool is to design and debug
+concepts, and then translate them to STGs, where they can be used with
+these tools.
 
 The latest version of this tool, and this manual, can be found [in the
 GitHub repo](https://github.com/tuura/concepts). Any bugs or issues
 found with the tool can be reported here. This tool is also distributed
-as a back-end tool for <span><spanstyle="font-variant:small-caps;">Workcraft</span></span>. This version
+as a back-end tool for orkcraft. This version
 of the concepts tool will be the latest version that works correctly
```
Comment (snowleopard) [inline doc/manual.md:36]: Orkcraft sounds great, but I'm not sure orks actually have any use for asynchronous concepts... :thinking:

Comment (jrbeaumont) [reply doc/manual.md:36]: They may do, how do we know? I don't know any Orks :(
I will change this just incase!

**Comment by jrbeaumont** [comment]
@snowleopard This should be fixed :)

**Comment by snowleopard** [comment]
Thanks @jrbeaumont! Shall I merge this now?

**Comment by jrbeaumont** [comment]
@snowleopard Yes please, Travis is being slow on macOS again.

**Comment by jrbeaumont** [comment]
@snowleopard: Wait no.

**Comment by snowleopard** [comment]
Oops, merged!

**Comment by jrbeaumont** [comment]
No problem, I'll send another PR.

**Comment by jrbeaumont** [comment]
My fault, sorry!

**Comment by snowleopard** [comment]
I've reverted the commit, so do send another PR

**Comment by jrbeaumont** [comment]
OK, thanks. Sorry!

**Pull request merged**